sample/weighted.go

```diff
@@ -0,0 +1,98 @@
+// Copyright ©2015 The gonum Authors. All rights reserved.
+// Use of this code is governed by a BSD-style
+// license that can be found in the LICENSE file
+
+package sample
+
+import "math/rand"
+
+// Weighted provides sampling without replacement from a collection of items with
+// weighted probabilites.
```

Comment:
I just thought, this is wrong - they are not probabilities, they are weights. How would you like me to phrase that?